src/react/SessionManager.js

```diff
@@ -0,0 +1,175 @@
+/*
+   This file is part of Astarte.
+
+   Copyright 2020 Ispirata Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+function urlToSchemalessString(url) {
+  return url.toString().split(/https?:\/\//)[1];
+}
+
+class SessionManager {
```

Comment:
Maybe it could be cleaner to have some outside functions to handle fetch/store session from localStorage.
Also, it seems you are saving redundant data inside this class; you could just have a `this.session` property and have property getters for `this.credentials` or `this.isLoggedIn`, along with encoder/decoder for `session` <=> `{ config, credentials, loginType }`

Reply:
This session object is the exact object that will saved in the storage. I save the credentials in the class because i don't want to de-serialize the object every time i need to access one property.
As for the load/store from the storage, i didn't want to over-complicate this PR, this will be handled in a different PR when we want to support different storage handlers (like cookies or indexDB)

Reply:
I'll write a single internal object for that, and encode the session object when required :+1: